src/lib.rs

```diff
@@ -432,4 +490,37 @@ impl<T: Clone + Copy> RbConsumer<T> for Consumer<T> {
         self.slots_free.notify_one();
         Some(cnt)
     }
+
+    fn read_blocking_timeout(&self, data: &mut [T], timeout: Duration) -> Result<Option<usize>> {
+        if data.is_empty() {
+            return Ok(None);
+        }
+        let guard = self.buf.lock().unwrap();
+        let buf = if self.inspector.is_empty() {
+            let (guard, result) = self.data_available.wait_timeout(guard, timeout).unwrap();
+            if result.timed_out() {
+                return Err(RbError::TimedOut);
+            }
+            guard
+        } else {
+            guard
+        };
```

Comment:
Same as for `write_blocking_timeout`, would be nice if we could share the logic.

Reply:
Pushed this too, thanks!